### PR TITLE
Fix error when changing beatmap and starting player at the same time

### LINKED ISSUE #3666: Starting a map gave an error and loaded another map
No idea how to reproduce this it happend random
[runtime.log](https://github.com/ppy/osu/files/2537544/runtime.log)

### PR DESCRIPTION
**Pull request by qwewqa**
Fixes #3666 

Should fix occasional errors when:
- Beatmap is changed with arrow keys while entering player
- Typing while entering player
- Clicking on a beatmap while entering player

### CODE REVIEW

File: osu.Game/Screens/Select/SongSelect.cs [hunk 1]
```diff
@@ -311,6 +311,18 @@ private void performUpdateSelected()
 
             void run()
             {
+                if (Beatmap.Disabled)
```
Comment (peppy) [inline osu.Game/Screens/Select/SongSelect.cs]: i really dislike this logic. can this nor just check if `Beatmap.Disabled` and not change selection?

Comment (qwewqa) [reply osu.Game/Screens/Select/SongSelect.cs]: It's scheduled with a delay so we can't check earlier. The display has already changed in the carousel if it was clicked but the actual active beatmap hasn't undated yet so entering a level will enter one different than the one indicated.

Comment (peppy) [reply osu.Game/Screens/Select/SongSelect.cs]: There has to be a better way to do this. Adding code like this is going in the wrong direction – it's piling more hacks on top of sub-optimal existing code.

**Comment by peppy** [comment]
This has since been fixed via other means.

**Pull request closed without merging**